templates/default/deploy-scripts/src/deployer.rs

```diff
@@ -0,0 +1,394 @@
+    /// given. If `None` is provided, 50000 energy is used as a default energy
+    /// value. An optional expiry time for the transaction can be given. If
+    /// `None` is provided, the local time + 300 seconds is used as a default
+    /// expiry time.
+    pub async fn update_contract(
+        &mut self,
+        update_payload: UpdateContractPayload,
+        energy: Option<GivenEnergy>,
+        expiry: Option<TransactionTime>,
+    ) -> Result<(TransactionHash, BlockItemSummary), Error> {
+        println!("\nUpdating contract....");
+
+        let nonce = self.get_nonce(self.key.address).await?;
+
+        if !nonce.all_final {
+            bail!("Nonce not final")
+        }
+
+        let payload = transactions::Payload::Update {
+            payload: update_payload,
+        };
+
+        let expiry = expiry.unwrap_or(TransactionTime::from_seconds(
+            (chrono::Utc::now().timestamp() + 300) as u64,
+        ));
+
+        let energy = energy.unwrap_or(GivenEnergy::Absolute(Energy { energy: 50000 }));
+
+        let tx = transactions::send::make_and_sign_transaction(
+            &*self.key,
+            self.key.address,
+            nonce.nonce,
+            expiry,
+            energy,
+            payload,
+        );
+        let bi = transactions::BlockItem::AccountTransaction(tx);
+
+        let tx_hash = self.client.clone().send_block_item(&bi).await?;
+
+        println!("Sent tx: {tx_hash}");
+
+        let (_, block_item) = self.client.wait_until_finalized(&tx_hash).await?;
+
+        self.check_outcome_of_update_transaction(&block_item)?;
+
+        println!("Transaction finalized: tx_hash={}", tx_hash,);
+
+        Ok((tx_hash, block_item))
+    }
+
+    /// A function to estimate the energy needed to send a transaction on the
+    /// chain.
+    ///
+    /// If successful, the transaction energy is returned by this function.
+    /// This function can be used to dry-run a transaction.
+    pub async fn estimate_energy(
+        &mut self,
+        payload: UpdateContractPayload,
+    ) -> Result<Energy, Error> {
+        let context =
+            ContractContext::new_from_payload(self.key.address, DEFAULT_INVOKE_ENERGY, payload);
+
+        let result = self
+            .client
+            .invoke_instance(&BlockIdentifier::LastFinal, &context)
+            .await?;
+
+        match result.response {
+            InvokeContractResult::Failure {
+                return_value,
+                reason,
+                used_energy,
+            } => bail!(format!(
+                "Contract invoke failed: {reason:?}, used_energy={used_energy}, return \
+                 value={return_value:?}"
+            )),
+            InvokeContractResult::Success {
+                return_value: _,
+                events: _,
+                used_energy,
+            } => Ok(used_energy),
+        }
+    }
+
+    /// A function to get the next nonce of the wallet account.
+    pub async fn get_nonce(
```

Comment:
We should use the `sequence_number` terminology since that's the terminology in the node API nowadays.

Reply:
The return value is called `AccountNonceResponse` and hence I would stick to the `nonce` terminology. In my opinion, an account nonce is clear what it is used for while a `sequence_number` is not so clear what it relates to.